x-pack/plugins/security_solution/cypress/integration/detection_alerts/alert_flyout.spec.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  ALERT_FLYOUT,
+  SUMMARY_VIEW_PREVALENCE_CELL,
+  SUMMARY_VIEW_INVESTIGATE_IN_TIMELINE_BUTTON,
+  INSIGHTS_RELATED_ALERTS_BY_SESSION,
+  INSIGHTS_INVESTIGATE_IN_TIMELINE_BUTTON,
+} from '../../screens/alerts_details';
+import { QUERY_TAB_BUTTON, TIMELINE_TITLE } from '../../screens/timeline';
+
+import { expandFirstAlert } from '../../tasks/alerts';
+import { openOverview } from '../../tasks/alerts_details';
+import { createCustomRuleEnabled } from '../../tasks/api_calls/rules';
+import { cleanKibana } from '../../tasks/common';
+import { waitForAlertsToPopulate } from '../../tasks/create_new_rule';
+import { esArchiverLoad, esArchiverUnload } from '../../tasks/es_archiver';
+import { login, visitWithoutDateRange } from '../../tasks/login';
+
+import { getNewRule } from '../../objects/rule';
+
+import { ALERTS_URL } from '../../urls/navigation';
+import { CLOSE_TIMELINE_BUTTON } from '../../screens/security_main';
+
+describe('Alert Flyout', () => {
+  before(() => {
+    cleanKibana();
+    esArchiverLoad('auditbeat');
+    login();
+    createCustomRuleEnabled(getNewRule(), 'rule1');
+    visitWithoutDateRange(ALERTS_URL);
+    waitForAlertsToPopulate();
+    expandFirstAlert();
+  });
+  after(() => {
+    esArchiverUnload('auditbeat');
+  });
+
+  afterEach(() => {
+    cy.get(CLOSE_TIMELINE_BUTTON).click();
+  });
+
+  it('Opens a new timeline investigation (from a prevalence field)', () => {
+    openOverview();
+
+    cy.get(SUMMARY_VIEW_PREVALENCE_CELL)
+      .first()
+      .invoke('text')
+      .then((alertCount) => {
+        // Click on the first button that lets us investigate in timeline
+        cy.get(ALERT_FLYOUT).find(SUMMARY_VIEW_INVESTIGATE_IN_TIMELINE_BUTTON).first().click();
```

Comment:
nit: Instead of adding comments, you can wrap that code into meaningful methods and add them to the `tasks` folder. In this way, everyone can easily reuse the code in the future if needed.